### ISSUE TITLE
allow customizing the branch

### ISSUE THREAD

**Issue by adrianjost** [adrianjost/files-sync-action#16]
---
_This issue has been automatically created by [todo-actions](https://github.com/apps/todo-actions) based on a TODO comment found in [src/git.js:50](https://github.com/adrianjost/files-sync-action/blob/master/src/git.js#L50). It will automatically be closed when the TODO comment is removed from the default branch (master)._

**Comment by adrianjost** [comment]
Should this be on a per repo basis or only a single option which is active for all `TARGET_REPOS` and the `SRC_REPO`?

**Comment by alvarosanchez** [comment]
I think the source branch will be `master` in the majority of the cases, but maybe a `SRC_BRANCH` param could be added.

Regarding the target branch(es), ideally one could define some regexs in `TARGET_BRANCHES`:

```yaml
TARGET_BRANCH: |-
  ^master$
  ^.*x$  # match 1.1.x, 1.2.x, etc
```

**Comment by adrianjost** [comment]
So you would like to see an option to sync files into multiple branches of the same repo?

**Comment by alvarosanchez** [comment]
Correct.

Our use case is that we use this action to sync GitHub Actions workflows, to do releases among other things. In our repos, apart from the master branch, we have maintenance releases for older versions like 1.1.x, 1.2.x, etc. We would like the actions to be sync'ed to those branches too.

**Comment by adrianjost** [comment]
@alvarosanchez Just as a quick update for you:
I do not have that much spare time at the moment, but this will be the next feature I implement in this project.

---

If someone would like to contribute with a PR, I think these are the steps that need to be implemented:
- fetch all branch names of each repo
- method to switch branches
- method to filter list of branches by regex
- update copy loop with an additional loop to switch between all matching branches of each repo.

All contributions are welcome, even if it is just a comment or issue with a small code-snippet :)

**Comment by alvarosanchez** [comment]
No worries.

Anyway, we have moved away from this action and implemented our own with bash scripts, since we needed more customizations.

Thanks anyway!

**Comment by adrianjost** [comment]
Thanks for letting me know :)

**Comment by adrianjost** [comment]
It's now implemented in v1.4.1 🎉🥳 
https://github.com/adrianjost/files-sync-action/releases/tag/v1.4.1

You can define multiple branches for a repo by simply defining it multiple times with different branch names.

`repoSlug:optionalBranchName`
